source/Purchases.brs

```diff
@@ -839,15 +854,49 @@ function _InternalPurchases(o = {} as object) as object
                     monthly: monthly,
                     availablePackages: availablePackages,
                 }
-                all_offerings.push(o)
-                if offering.identifier = current_offering_id
+                all_offerings[offering.identifier] = o
+                if is_current_offering
                     current_offering = o
                 end if
             end for
             return {
                 data: {
                     current: current_offering,
                     all: all_offerings,
+                    placements: placements,
+                    targeting: targeting,
+                    _offeringWithPlacement: sub(inputArgs = {}) as object
```

Comment:
I need to implement proper copy behavior, because BrightScript does not have a built-in way to copy objects.

Currently we're modifying the existing offering, but that would not be correct if the offering is first fetched with `currentOfferingForPlacement` and then via `all` and then a purchase happens.